CONTRIBUTING.md

```diff
@@ -0,0 +1,37 @@
+# Contributing to Rust IPFS
+
+Welcome, and thank you for your interest in contributing to Rust IPFS. Issues and pull requests are encouraged. As we work through the efforts described in the approved IPFS Dev Grant, we're looking for help in the following
+
+* Implementing endpoints and features not covered in the dev grant (See the [README](./README.md#roadmap))
+* Tests and CI for existing functionality
+* Examples
+* Documentation
+
+## First Principles
+
+1. Keep the build time small
+2. Aim for high (but not absolute) code coverage in testing
+3. Keep the dependency count low (use [`cargo-deps`](https://github.com/m-cat/cargo-deps))
```

Comment:
I am thinking there is likely a lot of work to do ahead and adding these principles to minimize the build time might complicate it. I feel that these are good principles except for the dependency count, which could transitively give you better build times but that might be better as a later optimization...? Building a dependencyless ipfs node is possible but community efforts should be leveraged, so I feel there is a conflict.

Otherwise these wouldn't matter, but I feel like if I approve these, I should make sure my next PR would probably at least more difficult to do. Not sure what to write here, or if anything should. Now that something has been offered, I feel like we should have something like this. Any other ideas?

Reply:
I'll remove the dependency count line for now and other people can chime in. We'll learn more as we go and actively collaborate